packages/sources/moore-hk/src/transport/trueusd.ts

```diff
@@ -0,0 +1,82 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { BaseEndpointTypes } from '../endpoint/trueusd'
+
+interface ResponseSchema {
+  accountName: string
+  totalTrust: number
+  totalToken: number
+  updatedAt: string
+  token: {
+    tokenName: string
+    totalTokenByChain: number
+  }[]
+  ripcord: boolean
+  ripcordDetails: string[]
+}
+
+export type HttpTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+
+export const httpTransport = new HttpTransport<HttpTransportTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      return {
+        params: [param],
+        request: {
+          baseURL: config.API_ENDPOINT,
+          url: '/chainlink/proof-of-reserves/TrueUSD',
+        },
+      }
+    })
+  },
+  parseResponse: (params, response) => {
+    return params.map((param) => {
+      // Return error if ripcord indicator true
+      if (response.data.ripcord) {
+        const message = `Ripcord indicator true. Details: ${response.data.ripcordDetails.join(
+          ', ',
+        )}`
+        return {
+          params: param,
+          response: {
+            errorMessage: message,
+            statusCode: 502,
+            timestamps: {
+              providerIndicatedTimeUnixMs: new Date(response.data.updatedAt).getTime(),
```

Comment:
nit to clean up parsing the indicated timestamp twice. We could parse it at the top and just set it here and below.